### PR TITLE
Milestone 2: Decorate A Class

### PR DESCRIPTION
**Pull request by tresorsawasawa**
Below is a list of tasks I completed in this PR:
- [ ] Created a `Nameable` class.
  - Implement a method called correct_name that will raise a NotImplementedError.
- [ ] Made `Person` class inherits from Nameable and implemented `correct_name` method returning the **name** attribute.
- [ ] Created a `BaseDecorator` class
  - Made sure that it inherits from `Nameable`.
  - In the constructor assign a nameable object from params to an instance variable.
  - Implement the `correct_name` method that returns the result of the `correct_name` method of the `@nameable`.
- [ ] Created a `CapitalizeDecorator` class
   - Made sure that it inherits from `BaseDecorator`.
   - Implement a method `correct_name` that capitalizes the output of` @nameable.correct_name`.
- [ ] Created a `TrimmerDecorator` class
   - Made sure that it inherits from `BaseDecorator`.
   - Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.
- [ ] Added the [main.rb](https://github.com/tresorsawasawa/OOP_School_Library/pull/2/files#diff-8c2db00221aa4d66a91598201c26bb6b695ccd79084d53d9783d74f615dba898) file that tests the decoration.

### CODE REVIEW

File: lib/nameable.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
+  end
+end
```
Comment (mateo951) [inline lib/nameable.rb:5]: Well done! You create Namable class with a method 'correct_name' that raises an error 🔥

File: lib/person.rb [hunk 2]
```diff
@@ -1,8 +1,11 @@
-class Person
+require_relative 'nameable'
+
+class Person < Nameable
```
Comment (mateo951) [inline lib/person.rb:3]: Nice 🔥 Your Person class inherits from Nameable!

**Review by mateo951 — APPROVED** [review]
### Hi @tresorsawasawa, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 😎 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by tresorsawasawa** [comment]
Hey @mateo951 

Thank You for the review :pray: 
I really appreciate :+1: :+1:

**Pull request merged**